npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb

```diff
@@ -395,6 +399,13 @@ def run_checker(path:, version:)
           run_npm_checker(path: path, version: version)
         end
 
+        def dependency_has_peer_dependencies?(version:)
+          command = "npm info #{dependency.name}@#{version} peerDependencies"
```

Comment:
Just having a quick poke in npm 7 lockfiles and looks like we have `peerDependencies` in there for all existing versions (not the case for npm 6), so if there's a lockfile we could check if any dependencies have `peerDependencies` defined and run this command for the new version and short-circuit.

https://github.com/dependabot/dependabot-core/blob/e5faa799769de92975587c8bb2b65c0c374c2f6d/npm_and_yarn/spec/fixtures/projects/npm7/peer_dependency/package-lock.json#L135-L137

Reply:
Yeah good call. It's _slightly_ tricky though, in that if you have an existing project and you upgrade by simply running `npm install` with npm 7, it won't actually add this `peerDependencies` info :/